src/main/java/org/sunbird/workflow/service/impl/UserProfileWfServiceImpl.java

```diff
@@ -64,11 +69,122 @@ public void updateUserProfile(WfRequest wfRequest) {
+		String endPoint = configuration.getUserProfileReadEndPoint().replace(Constants.USER_ID_VALUE,
 				wfRequest.getApplicationId());
-		builder.append(configuration.getHubServiceHost()).append(endPoint);
-		requestServiceImpl.fetchResult(builder, wfRequest.getUpdateFieldValues(), Map.class);
+		builder.append(configuration.getLmsServiceHost()).append(endPoint);
+		Object readResponse = requestServiceImpl.fetchResultUsingGet(builder);
+		Map<String, Object> readData = mapper.convertValue(readResponse,Map.class);
+		String existingDeptName = (String)((Map<String, Object>)((Map<String, Object>)((Map<String, Object>) readData.get("result")).get("response")).get("rootOrg")).get("orgName");
+		String updatedDeptName = wfRequest.getDeptName();
+		if(!existingDeptName.equals(updatedDeptName)){
+			Map<String, Object> migrateRequestWrapper = new HashMap<>();
+			Map<String, Object> migrateRequestObject = new HashMap<>();
+			migrateRequestWrapper.put(Constants.USER_ID, wfRequest.getApplicationId());
+			migrateRequestWrapper.put(Constants.CHANNEL, configuration.getHubRootOrg());
+			migrateRequestWrapper.put(Constants.FORCE_MIGRATION, true);
+			migrateRequestWrapper.put(Constants.SOFT_DELETE_OLD_GROUP,true );
+			migrateRequestWrapper.put(Constants.NOTIFY_MIGRATION,false );
+			migrateRequestObject.put(Constants.REQUEST, migrateRequestWrapper);
+			builder = new StringBuilder();
+			builder.append(configuration.getLmsServiceHost()).append(configuration.getUserProfileMigrateEndPoint());
+			HashMap<String, String> headersValue = new HashMap<>();
+			headersValue.put("Content-Type", "application/json");
+			Map<String, Object> migrateeUserApiResp = (Map<String, Object>) requestServiceImpl
+					.fetchResultUsingPatch(builder, migrateRequestObject, Map.class, headersValue);
+		}
+		Map<String, Object> profileDetails = (Map<String, Object>)((Map<String, Object>)((Map<String, Object>) readData.get("result")).get("response")).get("profileDetails");
+		Map<String, Object> updateRequest = updateRequestWithWF(wfRequest.getApplicationId(),wfRequest.getUpdateFieldValues(), profileDetails);
+		Map<String, Object> requestObject = new HashMap<>();
+		Map<String, Object> requestWrapper = new HashMap<>();
+		requestWrapper.put(Constants.USER_ID, wfRequest.getApplicationId());
+		requestWrapper.put(Constants.PROFILE_DETAILS, updateRequest);
+		requestObject.put(Constants.REQUEST, requestWrapper);
+		HashMap<String, String> headersValue = new HashMap<>();
+		headersValue.put("Content-Type", "application/json");
+		builder = new StringBuilder();
+		builder.append(configuration.getLmsServiceHost()).append(configuration.getUserProfileUpdateEndPoint());
+		Map<String, Object> updateUserApiResp = (Map<String, Object>) requestServiceImpl
+				.fetchResultUsingPost(builder, requestObject, Map.class, headersValue);
+	}
+
+
+	public Map<String, Object> updateRequestWithWF(String uuid, List<HashMap<String, Object>> requests,Map<String, Object> search) {
```

Comment:
the variable names are too general in this method and it's confusing... 
search -- existingProfileDetail
requests - wfRequestParamList
request - wfRequestParamObj
sf - updatedProfileElementObj
searchFields - existingProfileElementList
obj - existingProfileElement
toChange - updatedProfileElement